mdsuite/transformations/map_molecules.py

```diff
@@ -119,286 +118,178 @@ def _run_dependency_check(self):
             if not self.database.check_existence(path):
                 self.get_prop_through_transformation(sp_name, self.dependency)
 
-    def _update_type_dict(self, dictionary: dict, path_list: list, dimension: int):
+    def _prepare_mass_array(self, species: list) -> list:
         """
-        Update a type spec dictionary.
+        Prepare an array of atom masses for the scaling.
 
         Parameters
         ----------
-        dictionary : dict
-                Dictionary to append
-        path_list : list
-                List of paths for the dictionary
-        dimension : int
-                Dimension of the property
+        species : list
+                List of species to be loaded.
         Returns
         -------
-        type dict : dict
-                Dictionary for the type spec.
+        mass_array : list
+                A list of masses.
         """
-        for item in path_list:
-            dictionary[str.encode(item)] = tf.TensorSpec(
-                shape=(None, None, dimension), dtype=tf.float64
-            )
+        mass_array = []
+        for item in species:
+            mass_array.append(self.experiment.species[item]["mass"])
 
-        return dictionary
+        return mass_array
```

Comment:
Single line - could also be a property. I think it should be `get_mass_array` or the property `mass_array`because prepare indicated to me, that it changes some class attributes. 
return `[self.experiment.species[item]["mass"] for item in self.species]`

Reply:
Adapted in 24dc27a